--- cog-0.10.0/CMakeLists.txt.orig	2021-05-18 17:03:23.000000000 +0200
+++ cog-0.10.0/CMakeLists.txt	2021-08-17 18:55:07.355282549 +0200
@@ -159,7 +159,7 @@
     VERSION ${COGCORE_VERSION}
     SOVERSION ${COGCORE_VERSION_MAJOR}
 )
-target_link_libraries(cogcore PkgConfig::WEB_ENGINE PkgConfig::SOUP)
+target_link_libraries(cogcore PkgConfig::WEB_ENGINE PkgConfig::SOUP -ldl)
 target_compile_definitions(cogcore PRIVATE G_LOG_DOMAIN=\"Cog-Core\")
 if (HAS_WALL)
     target_compile_options(cogcore PUBLIC -Wall)
@@ -172,7 +172,7 @@
       target_compile_options(cog PUBLIC "-Wall")
     endif ()
     target_compile_definitions(cog PRIVATE G_LOG_DOMAIN=\"Cog\")
-    target_link_libraries(cog cogcore -ldl)
+    target_link_libraries(cog cogcore)
 
     add_executable(cogctl cogctl.c core/cog-utils.c)
     set_property(TARGET cogctl PROPERTY C_STANDARD 99)
